### PR TITLE
[mlir][debug] Support DICommonBlock.

### PR DESCRIPTION
**Pull request by abidh**
A COMMON block is a named area of memory that holds a collection of variables. Fortran subprograms may map the COMMON block memory area to a list of variables. A common block is represented in LLVM debug by DICommonBlock.

This PR adds support for this in MLIR. The changes are mostly mechanical apart from small change to access the DICompileUnit when the scope of the variable is DICommonBlock.

### CODE REVIEW

**Review by Dinistro — APPROVED** [review]
We usually also added LLVM dialect roundtrip tests for such additions. Apart from that, this is LGTM!

File: mlir/lib/Target/LLVMIR/ModuleTranslation.cpp [hunk 1]
```diff
@@ -1064,19 +1064,28 @@ LogicalResult ModuleTranslation::convertGlobals() {
...
+      // the DICompileUnit in which to add it.
+      // But there are cases where the scope of the global points does not
+      // directly point to the DICompileUnit and we have to do a bit more work
+      // to get to it. Some of those cases are:
+      //
+      // 1. For the languages that support modules, the scope hierarchy can be
+      // variable -> DIModule -> DICompileUnit
+      //
+      // 2. For the Fortran common block variable, the scope hierarchy looks
+      // like as follows:
```
Comment (gysit) [inline mlir/lib/Target/LLVMIR/ModuleTranslation.cpp:1076]:
```suggestion
      // 2. For the Fortran common block variable, the scope hierarchy can be
```
nit:

**Review by gysit — APPROVED** [review]
LGTM modulo nits!

**Comment by abidh** [comment]
> We usually also added LLVM dialect roundtrip tests for such additions. Apart from that, this is LGTM!

Thanks for the review. I was not sure where the round trip test go. But I have added one in `test/Dialect/LLVMIR/debuginfo.mlir`. Please let me know if that works.

**Review by Dinistro — APPROVED** [review]
LGTM, thanks for addressing the comments.

**Pull request merged**